packages/datadog-plugin-koa/test/index.spec.js

```diff
@@ -501,6 +501,329 @@ describe('Plugin', () => {
+
+            let span
+
+            app.use((ctx, next) => {
+              span = tracer.scope().active()
+              return tracer.scope().activate(null, () => next())
+            })
+
+            app.use(ctx => {
+              ctx.body = ''
+
+              try {
+                expect(tracer.scope().active()).to.equal(span).and.to.not.be.null
+                done()
+              } catch (e) {
+                done(e)
+              }
+            })
+
+            getPort().then(port => {
+              appListener = app.listen(port, 'localhost', () => {
+                axios.get(`http://localhost:${port}/user`)
+                  .catch(done)
+              })
+            })
+          })
+
+          withVersions(plugin, ['koa-router', '@koa/router'], (routerVersion, moduleName) => {
+            let Router
+
+            beforeEach(() => {
+              Router = require(`../../../versions/${moduleName}@${routerVersion}`).get()
+            })
+
+            it('should do automatic instrumentation on routers', done => {
+              const app = new Koa()
+              const router = new Router()
+
+              router.get('user', '/user/:id', function handle (ctx, next) {
+                ctx.body = ''
+              })
+
+              app
+                .use(router.routes())
+                .use(router.allowedMethods())
+
+              agent
+                .use(traces => {
+                  const spans = sort(traces[0])
+
+                  expect(spans).to.have.length(1)
+                  expect(spans[0]).to.have.property('resource', 'GET /user/:id')
+                  expect(spans[0].meta).to.have.property('http.url', `http://localhost:${port}/user/123`)
+                })
+                .then(done)
+                .catch(done)
+
+              appListener = app.listen(port, 'localhost', (e) => {
+                axios
+                  .get(`http://localhost:${port}/user/123`)
+                  .catch(done)
+              })
+            })
+
+            it('should not lose the route if next() is called', done => {
+              const app = new Koa()
+              const router = new Router()
+
+              router.get('/user/:id', (ctx, next) => {
+                ctx.body = ''
+                return next()
+              })
+
+              app
+                .use(router.routes())
+                .use(router.allowedMethods())
+
+              agent
+                .use(traces => {
+                  const spans = sort(traces[0])
+
+                  expect(spans[0]).to.have.property('resource', 'GET /user/:id')
+                })
+                .then(done)
+                .catch(done)
+
+              appListener = app.listen(port, 'localhost', (e) => {
+                axios
+                  .get(`http://localhost:${port}/user/123`)
+                  .catch(done)
+              })
+            })
+
+            it('should support nested routers', done => {
```

Comment:
There are some of these tests like this one that I'm not sure are relevant to the `middleware` option. In general I would expect a test that checks that no middleware span is created, and a test to check that the request scope is reactivated, and that's it. Maybe with the addition of testing error throwing, but even that I would expect to just work sine the scope is reactivated.

Can you clarify why the additional tests?

Reply:
when i first started this feature there was some additional complexity to the Koa router instrumentation that I didnt quite grok and just wanted to be sure wasn't getting messed with unintentionally. Ive reviewed these again and don't really think they're related at this point, so i agree and have removed these, although I did keep the 1 error handling test in there since there's a fair bit of this middleware code that overlaps with setting error metadata on the request span.